src/org/zalando/stups/kio/api.clj

```diff
@@ -44,21 +46,28 @@
 
 ;; applications
 
-(defn read-applications [{:keys [search]} request db]
+(defn read-applications
+  [{:keys [search modified_before modified_after]} request db]
   (u/require-internal-user request)
-  (if (nil? search)
-    (do
-      (log/debug "Read all applications.")
-      (-> (sql/cmd-read-applications {} {:connection db})
-          (sql/strip-prefixes)
-          (response)
-          (content-type-json)))
-    (do
-      (log/debug "Search in applications with term %s." search)
-      (-> (sql/cmd-search-applications {:searchquery search} {:connection db})
-          (sql/strip-prefixes)
-          (response)
-          (content-type-json)))))
+  (let [conn {:connection db}
+        apps (if (nil? search)
+                 (sql/cmd-read-applications {} conn)
+                 (sql/cmd-search-applications {:searchquery search} conn))]
+    (if (nil? search)
+        (log/debug "Read all applications.")
+        (log/debug "Search in applications with term %s." search))
+    (->> apps
```

Comment:
although I think the implementation is correct: Why not filter this stuff within the sql query?